### PR TITLE
Updates MDL controls to allow remote updates. Closes #65.

### LINKED ISSUE #65: Allow MDL components to be updated remotely.
Provide a reference to the MDL component, and call required methods as needed to update their state. This happens when a component is updated programmatically rather then physically via UI.

### PR DESCRIPTION
**Pull request by chriscox**

### CODE REVIEW

File: src/ui/controls/RadioListControl.tsx [hunk 1]
```diff
@@ -40,6 +40,15 @@ export class RadioListControl extends React.Component<IStringControlProps, void>
   }
 
   /** @override */
```
Comment (appsforartists) [inline src/ui/controls/RadioListControl.tsx]: Are the `@override`s necessary?  React components are expected to implement these methods.

File: src/ui/controls/RadioListControl.tsx [hunk 2]
```diff
@@ -40,6 +40,15 @@ export class RadioListControl extends React.Component<IStringControlProps, void>
...
+  componentDidUpdate() {
+    const { limitedToValues, selectedValue } = this.props.variable;
+    let index = limitedToValues.indexOf(selectedValue);
+    let id = `${CSS.RMX_RADIO_LIST_ITEM}-Theme-${index}`;
+    let materialRadio = this.refs[id]["MaterialRadio"];
+    materialRadio.check();
+  }
+
```
Comment (appsforartists) [inline src/ui/controls/RadioListControl.tsx:51]: The `this.refs[key]` syntax is deprecated:

https://facebook.github.io/react/docs/refs-and-the-dom.html

Please use the callback syntax instead.

**Review by appsforartists — CHANGES_REQUESTED** [review]

File: src/ui/controls/RadioListControl.tsx (hunk 1 above)
Comment (chriscox) [reply src/ui/controls/RadioListControl.tsx]: I've added these only for future-proofing compatibility with closure compiler per recommendation here: http://usejsdoc.org/tags-override.html

File: src/ui/controls/RadioListControl.tsx (hunk 2 above)
Comment (chriscox) [reply src/ui/controls/RadioListControl.tsx:51]: Done. PTAL.

**Review by chriscox — COMMENTED** [review]
Updated the refs to use callbacks.

File: src/ui/controls/SwitchControl.tsx [hunk 3]
```diff
@@ -39,7 +42,7 @@ export class SwitchControl extends React.Component<IBooleanControlProps, void> {
 
   /** @override */
   componentDidUpdate() {
-    let materialSwitch = this.refs[CSS.RMX_SWITCH]["MaterialSwitch"];
+    let materialSwitch = this.switchControl["MaterialRadio"];
```
Comment (appsforartists) [inline src/ui/controls/SwitchControl.tsx]: Where is MaterialRadio coming from?  Isn't this a DOM element?

Comment (chriscox) [reply src/ui/controls/SwitchControl.tsx]: Sorry I had previously fixed that misspelling.  Its changed to MaterialSwitch (not sure why its showing the old one here.

Comment (chriscox) [reply src/ui/controls/SwitchControl.tsx]: And yes, MDL attaches these data elements to the controls, and each one has a specific way to trigger events programmatically on it.

File: src/ui/controls/RadioListControl.tsx (hunk 1 above)
Comment (appsforartists) [reply src/ui/controls/RadioListControl.tsx]: But it's not overriding anything:

https://github.com/facebook/react/blob/c78464f8ea9a5b00ec80252d20a71a1482210e57/src/isomorphic/modern/class/ReactComponent.js

https://github.com/facebook/react/blob/3bc5595dfd6db741b19f17e7aef98b2c1e4be847/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L331

**Comment by appsforartists** [comment]
Are there docs somewhere for MaterialRadio?  I can't find it.

File: src/ui/controls/RadioListControl.tsx (hunk 1 above)
Comment (chriscox) [reply src/ui/controls/RadioListControl.tsx]: Done.  Interesting... thanks for pointing that out. Updated. And I'll create separate PR to flush out the unnecessary overrides in other the classes.

**Comment by chriscox** [comment]
Unfortunately the docs are somewhat limited. However I'm following the recommended technique.. which you can find documented in the slider readme as an example:
https://github.com/google/material-design-lite/tree/mdl-1.x/src/slider#configuration-options

See the last Note of that section.

**Review by appsforartists — APPROVED** [review]

**Pull request merged**